src/dynamics/mpas/dp_coupling.F90

```diff
@@ -82,19 +82,24 @@ subroutine d_p_coupling(phys_state, phys_tend, pbuf2d, dyn_out)
    integer, allocatable :: bpter(:,:)  ! offsets into block buffer for packing data
    integer, allocatable :: cpter(:,:)  ! offsets into chunk buffer for unpacking data
 
-   real(r8), allocatable:: pmid(:,:)   !mid-level pressure consisten with MPAS discrete state
+   real(r8), allocatable:: pmid(:,:)      !mid-level hydrostatic pressure consistent with MPAS discrete state
+   real(r8), allocatable:: pintdry(:,:)   !interface hydrostatic pressure consistent with MPAS discrete state
+   real(r8), allocatable:: pmiddry(:,:)   !mid-level hydrostatic dry pressure consistent with MPAS discrete state
 
    real(r8), allocatable, dimension(:) :: bbuffer, cbuffer ! transpose buffers
    integer :: ierr
    character(len=*), parameter :: subname = 'd_p_coupling'
    !----------------------------------------------------------------------------
 
+   compute_energy_diags=&
+        (hist_fld_active('SE_dDP').or.hist_fld_active('SE_dDM').or.hist_fld_active('SE_dPD').or.&
+         hist_fld_active('KE_dDP').or.hist_fld_active('KE_dDM').or.hist_fld_active('KE_dPD').or.&
+         hist_fld_active('WV_dDP').or.hist_fld_active('WV_dDM').or.hist_fld_active('WV_dPD'))
```

Comment:
They are explained in dyn_comp.F90. Should I repeat description here or we could set the logical in dyn_comp.F90?

Reply:
Also: these are energy diagnostics and not the energy fixer. The energy fixer is in physics.

Reply:
> Also: these are energy diagnostics and not the energy fixer. The energy fixer is in physics.

Ah yes, thanks for the clarifications!

> They are explained in dyn_comp.F90. Should I repeat description here or we could set the logical in dyn_comp.F90?

Hmm yes, I see that now. I think it might make most sense to set the logical in `dyn_comp.F90`. However I'm not sure if dyn_init in dyn_comp.F90 is the best to define these diagnostics fields. @jtruesdal do you know if these diagnostics should be defined in dyn_init? It looks like there's some fields that are defined in stepon.F90, but I'm not familiar with the purpose of that module.

Reply:
Most of the history fields specific to a parameterization are defined in the initialization routine.  It makes sense to me that would be dyn_init for the dycores.  We should probably move the addfld calls taking place in stepon_init to dyn_init.

Reply:
@jtruesdal moving the `addfld` calls into `dyn_init` sounds good to me, if you think it is good. I'm not familar with the `stepon` module. Do you know what its purpose is?

@PeterHjortLauritzen after some thought, I think setting `compute_energy_diags` in `dyn_init` makes the most sense. As, I'm assuming, that `compute_energy_diags` wont change throughout a run. i.e. the energy fixer will either be on or off. Is that assumption correct?

Reply:
Will do

Reply:
This is a logical for diagnostics and it will not impact energy fixer in physics.